core/runtime/src/main/java/io/quarkus/artemis/core/runtime/ArtemisRuntimeConfig.java

```diff
@@ -2,28 +2,44 @@
 
 import java.util.Optional;
 
+import io.quarkus.artemis.core.common.ArtemisUtil;
+import io.quarkus.runtime.annotations.ConfigGroup;
 import io.quarkus.runtime.annotations.ConfigItem;
-import io.quarkus.runtime.annotations.ConfigPhase;
-import io.quarkus.runtime.annotations.ConfigRoot;
 
-@ConfigRoot(name = "artemis", phase = ConfigPhase.RUN_TIME)
+@ConfigGroup
 public class ArtemisRuntimeConfig {
 
     /**
      * Artemis connection url
      */
     @ConfigItem
-    public String url;
+    Optional<String> url;
```

Comment:
url is not optional. By making it non-optional it saves you from checking later (as the only thing done is throw an exception) and if you make it non optional here, Quarkus will give a meaningful error.

Or is this needed for multiple connections?

Reply:
It is in deed. Right now, the "root" configuration "is-a" runtime configuration, i.e. "inherits" all attributes. But this means that the url on the root should not be required. Furthermore, we most probably do not want a default connection as soon as we have one named connection (unless the default connection is explicitly configured). See https://github.com/turing85/quarkus-artemis/blob/feature/20-add-named-connections/core/runtime/src/main/java/io/quarkus/artemis/core/runtime/ArtemisRuntimeConfigs.java#L15